src/main/java/dev/ecattez/shahmat/board/ChessGame.java

```diff
@@ -0,0 +1,188 @@
+package dev.ecattez.shahmat.board;
+
+import dev.ecattez.shahmat.board.move.Capture;
+import dev.ecattez.shahmat.board.move.EnPassant;
+import dev.ecattez.shahmat.board.move.Movement;
+import dev.ecattez.shahmat.board.move.MovementVisitor;
+import dev.ecattez.shahmat.board.move.MovingStrategy;
+import dev.ecattez.shahmat.board.move.MoveOnVacant;
+import dev.ecattez.shahmat.board.pawn.PawnPromotionAllowedPieces;
+import dev.ecattez.shahmat.event.PromotionProposed;
+import dev.ecattez.shahmat.board.pawn.PawnPromotionRankVisitor;
+import dev.ecattez.shahmat.command.Move;
+import dev.ecattez.shahmat.command.Promote;
+import dev.ecattez.shahmat.event.BoardEvent;
+import dev.ecattez.shahmat.event.PawnPromoted;
+import dev.ecattez.shahmat.event.PieceCaptured;
+import dev.ecattez.shahmat.event.PieceMoved;
+import dev.ecattez.shahmat.event.PiecePositioned;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class ChessGame {
+
+    private static final PieceFactory PIECE_FACTORY = new PieceBox(); // fixme: change it
```

Comment:
Change for what ?

Reply:
I'm thinking about the singleton pattern.